### PR TITLE
swupd.bash: Multiple improvements

### LINKED ISSUE #1520: Disable 3rd-party related flags from swupd.bash autocompletion when --disable-third-party
When swupd is built without 3rd-party support by using the `--disable-third-party`, the 3rd-party options should be somehow disabled from the `swupd.bash` autocompletion script.

Related to #1516

### PR DESCRIPTION
**Pull request by lebensterben**
- All options and subcommands are automatically generated from help messages.
- Used `_init_completion` to set $cur, $prev, $words variables, which are
  equivalent to $2, $3, $COMP_WORD, but far more readable.
  - It also allows possibility to use other functions provided in
    `/usr/share/bash-completion/bash_completion`, for future
- A minor change on getting current OS version via `swupd info --quiet` from
  reading `/var/lib/swupd/version`, which is non-readable to non-root users
  - Though MoM file is still non-readable to non-root users
- We only ship modern Bash, so the test on Bash version near the end of the
  script is removed

This also closes #1520

### CODE REVIEW

**Comment by lebensterben** [comment]
The implication for this commit is, there's virtually no need to update this script if you just added some flags.

**Comment by castulo** [comment]
Looks like this is also related to #1064

**Comment by otaviobp** [comment]
Pushed a commit fixing tests

File: swupd.bash [hunk 1]
```diff
@@ -18,167 +18,94 @@
...
+    fi
+  elif [[ $prev == "swupd" ]]; then
+    if [[ $cur == -* ]]; then
+      opts=$(_parse_help swupd)
+    else
+      opts="$subcmds"
+    fi
+  elif grep -q "${words[1]}" <<<"$subcmds"; then
+    if [[ $cur == -* ]]; then
+      # [[ $cur == -- ]] && printf "%s" "$prev"
```
[the remark this answers is not in the document]
Comment (lebensterben) [reply swupd.bash:66]: removed

File: swupd.bash [hunk 2]
```diff
@@ -18,167 +18,94 @@
...
+      return 1
+    fi
+  elif [[ $prev == "swupd" ]]; then
+    if [[ $cur == -* ]]; then
+      opts=$(_parse_help swupd)
+    else
+      opts="$subcmds"
+    fi
+  elif grep -q "${words[1]}" <<<"$subcmds"; then
+    if [[ $cur == -* ]]; then
```
[the remark this answers is not in the document]
Comment (lebensterben) [reply swupd.bash:65]: One alternative is to always show flags except for `bundle-add`, `bundle-remove`, and `hashdump`, which accepts non-option arguments and has alternative completions.

[the remark this answers is not in the document]
Comment (castulo) [reply swupd.bash:65]: 🤔I guess it is fine the way it is now, I mean, the only difference now is that you won't get the `--` autocompleted automatically if you do `swupd update <tab>`, right? you would need to write at least one `-` for it to work. If as you said @otaviobp that behavior is standard then that should be fine.

**Pull request merged**